scipy/_lib/_testutils.py

```diff
@@ -71,6 +73,73 @@ def __call__(self, label="fast", verbose=1, extra_argv=None, doctests=False,
         return (code == 0)
 
 
+class _TestPythranFunc:
+    '''
+    Inherit from this class to generate a bunch of relevant test cases
+    '''
+    ALL_INTEGER = [np.int8, np.int16, np.int32, np.int64]
```

Comment:
Last week we discussed whether it is useful/needed to add `intc`/`intp` here, and I thought the outcome was yes.